### DISCUSSION TITLE
RKES End Of Life and End Of Support for release 1.22 ,1.23 & 1.24

### DISCUSSION THREAD

**Discussion by himir11285** [rancher/rke2#3128] [General]
Please share the details of RKES End Of Life and End Of Support for release 1.22 ,1.23 & 1.24

I am not able to find in release 
https://github.com/rancher/rke2/releases?page=2

**Comment by brandond** [comment]
We produce RKE2 releases for every upstream Kubernetes release, and follow the same end-of-life schedule: https://kubernetes.io/releases/

**Comment by turkcankeskin** [comment]
Hi @brandond ,

Is what you said valid in 2024-2025?
Are you still following the LTS/EOL of kubernetes itself?

**Reply by brandond** [reply]
There is no LTS for Kubernetes. Barring extraordinary circumstances, we continue to put out releases only following upstream patches.

**Reply by turkcankeskin** [reply]
Hi @brandond ,

First of all, thanks for your answer.
I know the eol/lts logic of open source software here. But for example, I am currently using v1.27.12+rke2r1 / rancher: 2.8.3.

I also know that if a critical security vulnerability occurs in kubernetes or rke2, I should always follow the latest version.

But it may force me to always switch to the latest version. https://endoflife.date/kubernetes / https://endoflife.date/rancher In this case, are patches passed in intermediate versions? My question is, does rancher follow this? There may be bulk updates in between. I may need to check this.

**Reply by brandond** [reply]
Patch releases have whatever updates are available at the time they are released. We do not "re-release" old patches with new vulnerabilities addressed or issues fixed. If you want the latest updates, and the least vulnerabilities, use the latest patch release of an actively maintained minor version of Kubernetes.

For example, there are newer patch releases of both RKE2 and Rancher available to you. You don't have to upgrade to rke2 1.28 or rancher 2.9 (although you should), but you haven't even installed the latest patches for the minor versions you're CURRENTLY using.

**Reply by turkcankeskin** [reply]
Thanks so much for your answers, @brandond
